src/Processors/Transforms/AggregatingTransform.cpp

```diff
@@ -372,6 +375,11 @@ class ConvertingAggregatedToChunksTransform : public IProcessor
         return prepareTwoLevel();
     }
 
+    void onCancel() override
```

Comment:
`ConvertingAggregatedToChunksTransform` is stored inside `processors` field in `AggregatingTransform`, I think we should call `cancel` on each processor in `processors` in `AggregatingTransform::onCancel()`, otherwise this `onCancel` may be not called at all.

Reply:
If processors are not added to pipeline in `expandPipeline` of `AggregatingTransform` it is probably okay if we do not cancel them, because they did not do any work in `work()` function.